### ISSUE TITLE
tests failing due to scikit-learn update

### ISSUE THREAD

**Issue by wdevazelhes** [scikit-learn-contrib/metric-learn#127]
Some tests are failing in master, due to scikit-learn update v0.20.0. There are two types of tests that are failing: 

- Tests that uses the iris dataset: this is probably due to the fact that the iris dataset has changed in scikit learn v.0.20: cf http://scikit-learn.org/stable/whats_new.html#sklearn-datasets
For now maybe we can do a quickfix to pass the tests but maybe in the future we should replace these tests that test hard numerical values ? 

- Tests using sklearn's check_estimator: they fail for the check `check_fit2d_1sample` which was improved in v0.20 by checking the actual error message that is returned: see diff between lines 594-595 of v0.19.0 and lines 843-845 of v0.20 [here](https://github.com/scikit-learn/scikit-learn/compare/0.19.0...0.20.0) in Files changed, `sklearn/utils/estimator_checks.py` (the git diff didn't really worked here so these lines are not aligned). They can be fixed by setting `ensure_min_samples=2` in the checks that are done at fit, but we should probably think more about what are the minimum number of samples needed to make the algorithms work (since scikit-learn only tests this error is thrown for 1 sample, but maybe the algorithm won't work for 2 samples either and return the bad error message as before, which is not ideal...)

**Comment by bellet** [comment]
I think this has been solved, and a separate issue has been created for the question of setting minimal conditions to allow fit (#128)
